File: Module 19-Sorting&algo/02_Stableunstable.py
s = "DYIFZHZFHXEOHXXYUZ"

# Step 1: filter out only chars >= 'X'
ans = "".join([ch for ch in s if ch >= 'X'])
print("Filtered:", ans)

# Step 2: bubble sort (ascending order)
ans_list = list(ans)   # convert string to list for swapping
n = len(ans_list)

for i in range(n):
    swapped = False
    for j in range(0, n - 1 - i):   # bubble pass
        if ans_list[j] > ans_list[j + 1]:  # ascending order
            ans_list[j], ans_list[j + 1] = ans_list[j + 1], ans_list[j]
            swapped = True
    if not swapped:
        break  # stop early if already sorted

sorted_ans = "".join(ans_list)
print("Sorted:", sorted_ans)

File: Module 19-Sorting&algo/02_Stableunstable.cpp
#include<iostream>
#include<string>
using namespace std;

int main(){
    string str ="DYIFZHZFHXEOHXXYUZ";
    string ans = "";
    for (int i = 0;i<str.length();i++){
        if(str[i]>='X'){
            ans.push_back(str[i]);
        }
    }
    cout<<ans<<endl;
    for (int i = 0;i<ans.length();i++){
        bool check = false;
        
        for (int j = 0;j<ans.length()-i-1;j++){
            if(ans[j]<=ans[j+1]){
                swap(ans[j],ans[j+1]);
                check=true;
            }
        }
        if(check==false){
            break;
        }
    }
    cout<<ans;

}